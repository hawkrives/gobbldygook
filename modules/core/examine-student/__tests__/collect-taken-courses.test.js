import collectTakenCourses from '../collect-taken-courses'

describe('collectTakenCourses', () => {
	it('collects a list of all of the courses anywhere in this object which have the `_taken` property.', () => {
		const obj = {
			$type: 'course',
			$course: {
				department: [ 'ASIAN' ],
				number: 120,
			},
			_taken: true,
		}

		expect(collectTakenCourses(obj)).toMatchSnapshot()
	})

	it('can go down one layer deep', () => {
		const obj = {
			$type: 'boolean',
			$booleanType: 'or',
			$or: [
				{
					$type: 'course',
					$course: {
						department: [ 'ASIAN' ],
						number: 120,
					},
					_taken: true,
				},
				{
					$type: 'course',
					$course: {
						department: [ 'ASIAN' ],
						number: 120,
					},
				},
			],
		}

		expect(collectTakenCourses(obj)).toMatchSnapshot()
	})

	it('can go down many layers deep', () => {
		const obj = {
			$type: 'of',
			$count: { $operator: '$gte', $num: 3 },
			$of: [
				{
					$type: 'of',
					$count: { $operator: '$gte', $num: 1 },
					$of: [
						{
							$course: { department: [ 'CSCI' ], number: 120 },
							$type: 'course',
						},
						{
							$course: { department: [ 'CSCI' ], number: 121 },
							$type: 'course',
							_taken: true,
						},
						{
							$course: { department: [ 'CSCI' ], number: 122 },
							$type: 'course',
						},
					],
				},
				{
					$type: 'where',
					$count: { $operator: '$gte', $num: 2 },
					$where: {
						$key: 'gereq',
						$operator: '$eq',
						$type: 'qualification',
						$value: 'WRI',
					},
				},
				{
					$type: 'occurrence',
					$count: { $operator: '$gte', $num: 2 },
					$course: { department: [ 'CHEM' ], number: 121 },
				},
				{
					$type: 'of',
					$count: { $operator: '$gte', $num: 3 },
					$of: [
						{
							$course: { department: [ 'ART', 'ASIAN' ], number: 170 },
							$type: 'course',
							_taken: true,
						},
						{
							$course: { department: [ 'ART', 'ASIAN' ], number: 175 },
							$type: 'course',
							_taken: true,
						},
						{
							$course: { department: [ 'ART', 'ASIAN' ], number: 180 },
							$type: 'course',
						},
						{
							$course: { department: [ 'ART', 'ASIAN' ], number: 190 },
							$type: 'course',
							_taken: true,
						},
					],
				},
				{
					$type: 'of',
					$count: { $operator: '$gte', $num: 3 },
					$of: [
						{
							$course: { department: [ 'ASIAN' ], number: 210 },
							$type: 'course',
							_taken: true,
						},
						{
							$course: { department: [ 'ASIAN' ], number: 215 },
							$type: 'course',
							_taken: true,
						},
						{
							$course: { department: [ 'ASIAN' ], number: 220 },
							$type: 'course',
							_taken: true,
						},
					],
				},
			],
		}

		expect(collectTakenCourses(obj)).toMatchSnapshot()
	})
})
