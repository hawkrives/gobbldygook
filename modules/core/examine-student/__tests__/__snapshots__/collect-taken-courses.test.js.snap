exports[`collectTakenCourses can go down many layers deep 1`] = `
Array [
  Object {
    "department": Array [
      "CSCI",
    ],
    "number": 121,
  },
  Object {
    "department": Array [
      "ART",
      "ASIAN",
    ],
    "number": 170,
  },
  Object {
    "department": Array [
      "ART",
      "ASIAN",
    ],
    "number": 175,
  },
  Object {
    "department": Array [
      "ART",
      "ASIAN",
    ],
    "number": 190,
  },
  Object {
    "department": Array [
      "ASIAN",
    ],
    "number": 210,
  },
  Object {
    "department": Array [
      "ASIAN",
    ],
    "number": 215,
  },
  Object {
    "department": Array [
      "ASIAN",
    ],
    "number": 220,
  },
]
`;

exports[`collectTakenCourses can go down one layer deep 1`] = `
Array [
  Object {
    "department": Array [
      "ASIAN",
    ],
    "number": 120,
  },
]
`;

exports[`collectTakenCourses collects a list of all of the courses anywhere in this object which have the \`_taken\` property. 1`] = `
Array [
  Object {
    "department": Array [
      "ASIAN",
    ],
    "number": 120,
  },
]
`;
